### ISSUE TITLE
Will this use the FileReader to read?

### ISSUE THREAD

**Issue by PaulKinlan** [drufball/writable-files#2]
I would expect the API to prefer to use the `File` and `FileReader` API rather than a new custom `read` function.

**Comment by drufball** [comment]
I agree though that we'll want to reuse API's wherever possible. These are mostly just strawmen API ideas, rather than what I expect the final API to use. @inexorabletash @cmumford thoughts?

**Comment by kenchris** [comment]
Or maybe something like https://streams.spec.whatwg.org/#rs-class ReadableStream

**Comment by inexorabletash** [comment]
I assume we'll use File if possible. And while I'd love File to grow more usable reading methods like Request/Response's [Body](https://fetch.spec.whatwg.org/#body-mixin) that's probably out of scope, so I would assume File/FileReader are there.

**Comment by inexorabletash** [comment]
One plausible API sketch is here: https://gist.github.com/petkaantonov/37a9c64ea2bbd1aa3ca5

**Comment by addyosmani** [comment]
> I assume we'll use File if possible. And while I'd love File to grow more usable reading methods like Request/Response's Body that's probably out of scope, so I would assume File/FileReader are there.

This sounds like a reasonable start to me.
